ranker/shared/ranker_obj.py

```diff
@@ -278,24 +306,44 @@ def get_rgraph(self, answer):
                         redges.append(edge)
 
         return rnodes, redges
+    
 
+    
+def path_collapse(weighted_graph, probe):
+    if probe[0] == probe[1]:
+        return 1
+    if probe[0]>probe[1]:
+        probe = (probe[1],probe[0])
+    parallel_steps = [(i,w) for i, w in enumerate(weighted_graph[probe[0],:]) if w > 0]
+    parallel_steps_transpose = [(i,w) for i, w in enumerate(weighted_graph[:,probe[0]]) if w > 0]
+    parallel_steps = parallel_steps+parallel_steps_transpose
+    parallel_parts = []
+    weight_graph_temp = weighted_graph.copy()
+    if parallel_steps:
+        #this step takes out parallel connections as paths when they get sent to the recursion
+        for step in parallel_steps:
+            weight_graph_temp[probe[0],step[0]] = 0.
+            weight_graph_temp[step[0],probe[0]] = 0.
+        for step in parallel_steps:
+            new_p = (step[0], probe[1])
+            parallel_parts.append(series_combine([step[1], path_collapse(weight_graph_temp, new_p)]))
+            
+        out = parallel_combine(parallel_parts)
+    else:
+        return 0 #no connections. return 1
```

Comment:
return 1 or 0?

Reply:
Comment was not changed previously. Now fixed.